src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,272 @@
+into the Lie Algebra, i.e. performing a matrix logarithm beforehand.
+"""
+struct LogEuclideanMetric <: Metric end
+
+@doc doc"""
+    distance(P,x,y)
+
+computes the distance on the [`SymmetricPositiveDefinite`](@ref) manifold between `x` and `y`,
+which defaults to the [`LinearAffineMetric`](@ref) induces distance.
+"""
+distance(P::SymmetricPositiveDefinite{N},x,y) where N = distance(MetricManifold(P,LinearAffineMetric()),x,y)
+@doc doc"""
+    distance(lP,x,y)
+
+computes the distance on the [`SymmetricPositiveDefinite`](@ref) manifold between `x` and `y`,
+as a [`MetricManifold`](@ref) with [`LinearAffineMetric`](@ref). The formula reads
+
+```math
+d_{\mathcal P(n)}(x,y) = \lVert \operatorname{Log}(x^{-\frac{1}{2}}yx^{-\frac{1}{2}})\rVert.
+```
+"""
+function distance(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric},x,y) where N
+    s = real.( eigen( x,y ).values )
+    return any(s .<= eps() ) ? 0 : sqrt(  sum( abs.(log.(s)).^2 )  )
+end
+
+@doc doc"""
+    inner(P,x,v,w)
+
+compute the inner product of `v`, `w` in the tangent space of `x` on the [`SymmetricPositiveDefinite`](@ref)
+manifold `P`, which defaults to the [`LinearAffineMetric`](@ref).
+"""
+inner(P::SymmetricPositiveDefinite{N}, x, w, v) where N = inner(MetricManifold(P,LinearAffineMetric()),x,w,v)
+@doc doc"""
+    inner(P,x,v,w)
+
+compute the inner product of `v`, `w` in the tangent space of `x` on the [`SymmetricPositiveDefinite`](@ref)
+manifold `P`, as a [`MetricManifold`](@ref) with [`LinearAffineMetric`](@ref). The formula reads
+
+```math
+( v, w)_x = \operatorname{tr}(x^{-1}\xi x^{-1}\nu ),
+```
+"""
+function inner(P::MetricManifold{SymmetricPositiveDefinite{N}, LinearAffineMetric}, x, w, v) where N
+    F = factorize(x)
+    return tr( (w / F) * (v / F ))
+end
+@doc doc"""
+    exp!(P,y,x,v)
+
+compute the exponential map from `x` with tangent vector `v` on the [`SymmetricPositiveDefinite`](@ref)
+manifold with its default metric, [`LinearAffineMetric`](@ref) and modify `y`.
+"""
+exp!(P::SymmetricPositiveDefinite{N},y,x,v) where N = exp!(MetricManifold(P,LinearAffineMetric()),y,x,v)
+@doc doc"""
+    exp!(P,y,x,v)
+
+compute the exponential map from `x` with tangent vector `v` on the [`SymmetricPositiveDefinite`](@ref)
+as a [`MetricManifold`](@ref) with [`LinearAffineMetric`](@ref) and modify `y`. The formula reads
+
+```math
+    \exp_x v = x^{\frac{1}{2}}\operatorname{Exp}(x^{-\frac{1}{2}} v x^{-\frac{1}{2}})x^{\frac{1}{2}},
+```
+where $\operatorname{Exp}$ denotes to the matrix exponential.
+"""
+function exp!(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}, y, x, v) where N
+    e = eigen(Symmetric(x))
+    U = e.vectors
+    S = e.values
+    Ssqrt = Diagonal( sqrt.(S) )
+    SsqrtInv = Diagonal( 1 ./ sqrt.(S) )
+    xSqrt = U*Ssqrt*transpose(U);
+    xSqrtInv = U*SsqrtInv*transpose(U)
+    T = xSqrtInv * v * xSqrtInv
+    eig1 = eigen( T ) # numerical stabilization
```

Comment:
`xSqrt`, `xSqrtInv`, and `T` are all supposed to be symmetric, so I'd wrap them as such.